modules/fundamental/src/purl/service/mod.rs

```diff
@@ -230,41 +232,90 @@ impl PurlService {
         }
     }
 
-    pub async fn purl_by_purl<C: ConnectionTrait>(
+    #[instrument(skip(self, connection), err(level=tracing::Level::INFO))]
+    pub async fn fetch_purl_details<C: ConnectionTrait>(
         &self,
-        purl: &Purl,
+        identifiers: &[String],
```

Comment:
I think this is still valid:

`&[String]` is an immutable reference to a type considered for providing mutable string processing. Most likely, you're ok with just using `&str` and thus `&[&str]`.

However, you can also use a trait, and let the compiler figure it out: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=76815c841727331234f2e7bf437bda44

```rust
fn do_it(i: impl IntoIterator<Item=impl AsRef<str>>) {
    for i in i {
        println!("{}", i.as_ref());
    }
}

fn main() {
    do_it(["foo", "bar"]);
    do_it(["foo".to_string(), "bar".to_string()]);
    do_it(vec!["foo".to_string(), "bar".to_string()]);
}
```

Which basically says: give me something that I can turn into an iterator which gives me something that can turn into `&str`.